sys/net/application_layer/coap/gcoap.c

```diff
@@ -302,7 +302,7 @@ static void _find_resource(coap_pkt_t *pdu, coap_resource_t **resource_ptr,
 static ssize_t _finish_pdu(coap_pkt_t *pdu, uint8_t *buf, size_t len)
 {
     ssize_t hdr_len = _write_options(pdu, buf, len);
-    DEBUG("gcoap: header length: %u\n", hdr_len);
+    DEBUG("gcoap: header length: %i\n", (int)hdr_len);
```

Comment:
actually I think the error handling in `_write_options` is _suboptimal_. I mean the function actually returns `size_t` but also some negative error code, plus this cast its _ugly_. IMHO verify that the url starts with `/` should be done somewhere else.